examples/pghostname_zig/src/main.zig

```diff
@@ -0,0 +1,28 @@
+const std = @import("std");
+const pgzx = @import("pgzx");
+
+comptime {
+    pgzx.PG_MODULE_MAGIC();
+
+    pgzx.PG_EXPORT(@This());
+}
+
+pub fn pghostname_zig() ![]const u8 {
```

Comment:
nit: change the return type to `[:0]const u8`. By returning `[]const u8` you actually have implicitly casted the `[:0]const u8` from `dupeZ` into a `[]const u8`. As postgres requires proper zero terminated C strings we will implicitly allocate and copy the string again once your function returns. You can prevent the extra allocation and copying by returning `[:0]const u8`.

Reply:
Thanks, this is done.